app/models/user.rb

```diff
@@ -0,0 +1,21 @@
+class User < ApplicationRecord
+  before_create :add_avatar_url
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :username,
+            presence: true,
+            uniqueness: { case_sensitive: false },
+            format: {
+              with: /\A[a-zA-Z0-9_.-]*\z/,
+              message: 'only allows letters, numbers, underscores, periods, and dashes'
+            }
+
+  private
+
+  def add_avatar_url
+    self.avatar_url = "https://api.dicebear.com/6.x/bottts/svg?seed=#{self.username}"
```

Comment:
Not sure about this, but if username has periods, does that break this URL you are building? Maybe you can do `CGI.escape(username)` to escape any character that is not valid for a URL

```suggestion
    self.avatar_url = "https://api.dicebear.com/6.x/bottts/svg?seed=#{CGI.escape(username)}"
```

Reply:
no no, it doesn't break it, the url identifies right the seed

<img width="1218" alt="image" src="https://github.com/ImMPrada/chat-in-rails/assets/26731448/1dd436f4-af64-4219-a64d-0da15257d8b4">

Reply:
cool, anyways I guess it's still safe to escape the username to make sure nothing breaks the URL and your user creation. If an error happens on this callback the user can't sign up and there is no clear error boundary here, so it will just block the user from using the app

Reply:
good talk, done!